console.rst

```diff
@@ -101,6 +101,10 @@ messages to the console)::
             '',
         ]);
 
+        // the value returned by someMethod() can be an iterator that
+        // generates and returns the messages with the 'yield' PHP keyword
+        $output->writeln($this->someMethod());
```

Comment:
Would it be an idea to mention the Iterator? http://php.net/manual/en/class.iterator.php Could hint towards yield support next to that, but yield is a more complex concept than just an iterator